compiler_expr.go

```diff
@@ -2327,6 +2336,12 @@ func (c *compiler) compileArrowFunctionLiteral(v *ast.ArrowFunctionLiteral) *com
 
 func (c *compiler) emitLoadThis() {
 	b, eval := c.scope.lookupThis()
+
+	if c.module != nil && c.scope.outer != nil && c.scope.outer.outer == nil { // modules don't have this defined
+		// TODO maybe just add isTopLevel and rewrite the rest of the code
```

Comment:
Is it expected?

Reply:
I am going to try to do this in the next weeks, but AFAIR it was pretty hard to do this.

A bunch of other similar stuff also have ... strange ways of figuring out what is happening through the codebase, in part because the specification has some really cool corner cases.

But I will try to look into this specifically.

Reply:
I would like to leave this as a note on a thing I will like to refactor,  but I don't think it is going to be easy, so given the time pressure I will prefer to leave it for now.